### PR TITLE
(PDB-2927) Update PuppetDB doc references for 4.2

### PR DESCRIPTION
**Pull request by senior**

### CODE REVIEW

File: source/_config.yml [hunk 1]
```diff
@@ -58,6 +58,7 @@ symlink_latest:
   - facter
   - pe
 lock_latest:
+  - puppetdb 4.1
```
Comment (nfagerlund) [inline source/_config.yml]: Syntax error. The value of lock_latest is a YAML hash, not an array of strings.

**Comment by nfagerlund** [comment]
When does your team need this change merged?

File: source/_config.yml (hunk 1 above)
Comment (senior) [inline source/_config.yml]: So what do I need to change it to?

**Comment by senior** [comment]
@nfagerlund it can go anytime, we'll probably get the release out on Monday, which I'll follow this PR up with another one that unpins the latest version.

File: source/_config.yml (hunk 1 above)
Comment (senior) [inline source/_config.yml]: Looks like maybe I swap the [dash for a colon](https://github.com/puppetlabs/puppet-docs/pull/625/files#diff-e6422af2c5a3ea680839278540cfdf29L58)? I'll make the change.

**Comment by nfagerlund** [comment]
Yup, yaml hashes are `key: value`, so that fixed it. I'll merge.

**Pull request merged**